Description: Patch to use a shared enet library, generate binaries without 32 or 64 in the names, 
             and install in the proper directory
Author: Scott Howard <showard314@gmail.com>
Index: sandboxgamemaker-2.8.1/src/Makefile
===================================================================
--- sandboxgamemaker-2.8.1.orig/src/Makefile	2012-11-18 15:02:36.000000000 +0100
+++ sandboxgamemaker-2.8.1/src/Makefile	2012-11-18 15:08:01.142653263 +0100
@@ -7,7 +7,7 @@
 PLATFORM= $(shell uname -s)
 PLATFORM_PREFIX=sandbox
 
-INCLUDES= -Ishared -Iengine -Ienet/include
+INCLUDES= -Ishared -Iengine
 
 STRIP=
 ifeq (,$(findstring -g,$(CXXFLAGS)))
@@ -24,7 +24,7 @@
 CLIENT_LIBS= -mwindows -Llib -lSDL -lSDL_image -lSDL_mixer -lzdll -lopengl32 -lenet -lws2_32 -lwinmm
 else
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/X11R6/include `sdl-config --cflags`
-CLIENT_LIBS= -Lenet/.libs -lenet -L/usr/X11R6/lib `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL -lX11
+CLIENT_LIBS= -lenet -L/usr/X11R6/lib `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL -lX11
 endif
 ifeq ($(PLATFORM),Linux)
 CLIENT_LIBS+= -lrt
@@ -175,7 +175,7 @@
 MASTER_LIBS= -Llib -lzdll -lenet -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -Lenet/.libs -lenet -lz
+SERVER_LIBS= -lenet -lz
 MASTER_LIBS= $(SERVER_LIBS)
 endif
 
@@ -224,15 +224,6 @@
 
 all: client server
 
-enet/Makefile:
-	cd enet; ./configure --enable-shared=no --enable-static=yes
-
-libenet: enet/Makefile
-	$(MAKE)	-C enet/ all
-
-clean-enet: enet/Makefile
-	$(MAKE) -C enet/ clean
-
 clean:
 	-$(RM) $(SERVER_OBJS) $(FPSSERVER_OBJS) $(MOVIESERVER_OBJS) $(CLIENT_PCH) $(CLIENT_OBJS) $(SSPCLIENT_OBJS) $(RPGCLIENT_OBJS) $(FPSCLIENT_OBJS) $(MOVIECLIENT_OBJS) $(KRSCLIENT_OBJS) $(PZLCLIENT_OBJS) $(LAUNCHER_OBJS) $(MASTER_OBJS) $(PLATFORM_PREFIX)_server_* $(PLATFORM_PREFIX)_client_*
 
@@ -316,40 +307,42 @@
 install: all
 else
 
-client-fps:libenet $(CLIENT_OBJS) $(FPSCLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_$(MACHINENAME)_fps $(CLIENT_OBJS) $(FPSCLIENT_OBJS) $(CLIENT_LIBS)
+client-fps: $(CLIENT_OBJS) $(FPSCLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_fps $(CLIENT_OBJS) $(FPSCLIENT_OBJS) $(CLIENT_LIBS)
 
-client-ssp:libenet $(CLIENT_OBJS) $(SSPCLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_$(MACHINENAME)_ssp $(CLIENT_OBJS) $(SSPCLIENT_OBJS) $(CLIENT_LIBS)
+client-ssp: $(CLIENT_OBJS) $(SSPCLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_ssp $(CLIENT_OBJS) $(SSPCLIENT_OBJS) $(CLIENT_LIBS)
 
-client-rpg:libenet $(CLIENT_OBJS) $(RPGCLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_$(MACHINENAME)_rpg $(CLIENT_OBJS) $(RPGCLIENT_OBJS) $(CLIENT_LIBS)
+client-rpg: $(CLIENT_OBJS) $(RPGCLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_rpg $(CLIENT_OBJS) $(RPGCLIENT_OBJS) $(CLIENT_LIBS)
 
-client-movie:libenet $(CLIENT_OBJS) $(MOVIECLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_$(MACHINENAME)_movie $(CLIENT_OBJS) $(MOVIECLIENT_OBJS) $(CLIENT_LIBS)
+client-movie: $(CLIENT_OBJS) $(MOVIECLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_movie $(CLIENT_OBJS) $(MOVIECLIENT_OBJS) $(CLIENT_LIBS)
 
-client-krs:libenet $(CLIENT_OBJS) $(KRSCLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_$(MACHINENAME)_krs $(CLIENT_OBJS) $(KRSCLIENT_OBJS) $(CLIENT_LIBS)
+client-krs: $(CLIENT_OBJS) $(KRSCLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_krs $(CLIENT_OBJS) $(KRSCLIENT_OBJS) $(CLIENT_LIBS)
 
-client-pzl:libenet $(CLIENT_OBJS) $(PZLCLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_$(MACHINENAME)_pzl $(CLIENT_OBJS) $(PZLCLIENT_OBJS) $(CLIENT_LIBS)
+client-pzl: $(CLIENT_OBJS) $(PZLCLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_client_pzl $(CLIENT_OBJS) $(PZLCLIENT_OBJS) $(CLIENT_LIBS)
 
-server-fps: libenet $(SERVER_OBJS) $(FPSSERVER_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_server_$(MACHINENAME)_fps $(SERVER_OBJS) $(FPSSERVER_OBJS) $(SERVER_LIBS)
+server-fps: $(SERVER_OBJS) $(FPSSERVER_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_server_fps $(SERVER_OBJS) $(FPSSERVER_OBJS) $(SERVER_LIBS)
 
-server-movie: libenet $(SERVER_OBJS) $(MOVIESERVER_OBJS)
-	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_server_$(MACHINENAME)_movie $(SERVER_OBJS) $(MOVIESERVER_OBJS) $(SERVER_LIBS)
+server-movie: $(SERVER_OBJS) $(MOVIESERVER_OBJS)
+	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_server_movie $(SERVER_OBJS) $(MOVIESERVER_OBJS) $(SERVER_LIBS)
 
-master: libenet $(MASTER_OBJS)
+master: $(MASTER_OBJS)
 	$(CXX) $(CXXFLAGS) -o $(PLATFORM_PREFIX)_master $(MASTER_OBJS) $(MASTER_LIBS)
 
 
 install: all
 	cp $(PLATFORM_PREFIX)_* ../bin/
 ifneq (,$(STRIP))
-	strip ../bin/$(PLATFORM_PREFIX)_??????_$(MACHINENAME)*
+
 endif
-	chmod +x ../bin/$(PLATFORM_PREFIX)_??????_$(MACHINENAME)*
+	chmod +x ../bin/$(PLATFORM_PREFIX)_??????*
+	install -d $(DESTDIR)/usr/lib/sandboxgamemaker
+	install ../bin/$(PLATFORM_PREFIX)_??????_* $(DESTDIR)/usr/lib/sandboxgamemaker
 endif
 
 shared/cube2font.o: shared/cube2font.c
